test/src/unit-cppapi-array.cc

```diff
@@ -1421,6 +1427,49 @@ TEST_CASE(
     vfs.remove_dir(array_name);
 }
 
+TEST_CASE("RootPath", "[cppapi][config][rootpath]") {
+  std::string array_name = "test_array";
+
+#ifdef _WIN32
+  std::string root_path = tiledb::sm::Win::current_dir() + "\\test_root_path";
+#else
+  std::string root_path = std::string("file://") +
+                          tiledb::sm::Posix::current_dir() + "/test_root_path";
+#endif
+
+  tiledb::Config cfg;
+  cfg.set("vfs.local_root_path", root_path);
+  tiledb::Context ctx(cfg);
+  tiledb::VFS vfs(ctx);
+
+  if (vfs.is_dir(array_name)) {
+    vfs.remove_dir(array_name);
+  }
+
+  vfs.create_dir(root_path);
+
+  tiledb::Domain domain(ctx);
+  domain.add_dimension(tiledb::Dimension::create<int>(ctx, "d", {{0, 3}}, 1));
+  tiledb::ArraySchema schema(ctx, TILEDB_DENSE);
+  schema.set_domain(domain);
+  schema.add_attribute(tiledb::Attribute::create<int>(ctx, "a"));
+  tiledb::Array::create(array_name, schema);
+
+  tiledb::Array array(ctx, array_name, TILEDB_READ);
+  REQUIRE(array.config().get("vfs.local_root_path") == root_path);
+
+  std::string array_uri = array.uri();
+  array.close();
+
```

Comment:
Think it would be good to add in some portion of the following (and maybe more APIs should be covered too?), whichever is actually appropriate:
```
  CHECK_NOTHROW(array.vacuum(ctx, array_name));
  CHECK_NOTHROW(array.consolidate(ctx, array_name));
  CHECK_NOTHROW(array.vacuum(ctx, array_uri));
  CHECK_NOTHROW(array.consolidate(ctx, array_uri));
  CHECK_????(array.whatelse(ctx, array_uri_or_name));
```
I suppose with the removal/absence of make_uri() [from storage_manager routines] the array_uri versions are applicable, whereas with the make_uri() in play the array_name versions would probably have been correct.
I would have preferred to see the make_uri() functionality left, don't know why you removed it, as its absence will likely mean any user's finding a use for the root_path functionality will probably have to modify many more pieces of code to extract the root_path based .uri() from the array, whereas with it, seems it would have been enough just to modify one location, setting it into the config which would likely already be shared in common by code working with the same array calling its varied methods.

Reply:
We had three versions of setting root directory. The first version is a global setting. The second is constrained in a context. The current version has minimal effects(only for opening an array). Since we didn't get further information about the customer's requirements. we would like to go with the version with minimal effects. If we want to apply the rooting directory in a context in future, it is easy to add those things back.

Reply:
Ok.